[wip] move flags to c10

Yangqing opened the pull request:
still influx.

ezyang commented:
Do you think it's possible to consider a design where we don't have to have two builds of c10, one with glog/gflags, and one without? It would be much nicer if we could provide a *single* binary distribution that works both for people who do want to use glog/gflags, and those that don't.

Yangqing committed:
move flags to c10 (#12144)

Summary:
still influx.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12144

Differential Revision: D10140176

Pulled By: Yangqing

fbshipit-source-id: 13334e085a568d321cf9d96ff8464edff56eacc3

smessmer commented:
added some comments in Phabricator
